lib/kernel.ts

```diff
@@ -155,64 +185,44 @@ const getSelected = (schema: JsonSchema): { links: any; properties: any } => {
 	};
 };
 
-const rectifySelected = (
-	selected: any,
-	filteredSelected: { [x: string]: any },
-) => {
-	for (const [key, value] of Object.entries(selected)) {
-		if (key in filteredSelected) {
-			rectifySelected(value, filteredSelected[key]);
+// TODO: rename with something more descriptive.
+const rectifySelectObject = (
+	selectObject: SelectObject,
+	authorizedSelectObject: SelectObject,
+): SelectObject => {
+	const result = { ...selectObject };
+
+	for (const [key, value] of Object.entries(result)) {
+		if (key in authorizedSelectObject) {
+			rectifySelectObject(value, (authorizedSelectObject as any)[key]);
```

Comment:
Cast shouldn't be needed. Maybe the definition of `SelectObject` is not recursive?

Reply:
```
Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'SelectObject'.
  No index signature with a parameter of type 'string' was found on type 'SelectObject'.ts(7053)
```

Reply:
Ok ya looks like the definition is not right

Reply:
It's basically just a recursive dictionary of dictionaries

Reply:
You mean the `SelectObject` definition? This is how it currently looks like:
```
// TS-TODO: strongly type this object
export interface SelectObject {
	type?: { [key: string]: any };
	data?: { [key: string]: any };
	id?: { [key: string]: any };
	slug?: { [key: string]: any };
	links?: { [key: string]: any };
	properties?: { [key: string]: any };
}
```

Reply:
Ya I get why it's like that but it should be just

```js
export interface SelectObject {
  [key: string]: SelectObject;
}
```

Can you make that change?